File: examples/pairing-login.ts
import { Wazeko } from "../packages/wazeko/src/index.js";
import { PairingCodeEvent, ConnectionState } from "../packages/wazeko-types/src/index.js";

async function main() {
  const client = Wazeko.builder()
    .authStore("./auth_pairing")
    .pairingPhoneNumber("6281234567890")
    .build();

  client.on("pairing.code", (codeEvent: PairingCodeEvent) => {
    console.log(`Enter this code on WhatsApp: ${codeEvent.code}`);
  });

  client.on("connection.update", (state: ConnectionState) => {
    console.log(`Connection state: ${state}`);
  });

  await client.connect();
}

main().catch(console.error);

File: crates/wazeko/examples/pairing_login.rs
use wazeko::{Event, Wazeko};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    tracing_subscriber::fmt::init();

    let client = Wazeko::builder()
        .auth_store("./auth_pairing_demo")
        .pairing_phone_number("6281234567890")
        .build();

    client.connect().await?;

    let mut events = client.events();
    while let Some(event) = events.recv().await {
        match event {
            Event::PairingCode(code_event) => {
                println!("Got Pairing Code: {}", code_event.code);
            }
            Event::ConnectionUpdate(state) => {
                println!("Connection state: {state:?}");
            }
            _ => {}
        }
    }

    Ok(())
}
